packages/spartan/extra/gen_resolvconf.py

```diff
@@ -69,7 +69,20 @@ def check_server(addr):
 
 # If Spartan is not up, fall back, and insert the upstreams
 else:
-    fallback_servers = os.environ['RESOLVERS'].split(',')
+    fallback_servers = []
+
+    # Resolvconf does not support custom ports, skip if not default
+    for ns in os.environ['RESOLVERS'].split(','):
+        ip, separator, port = ns.rpartition(':')
+        if not separator:
+            fallback_servers.append(ns)
+            continue
+        if port == "53":
+            fallback_servers.append(ip)
```

Comment:
```
>>> "1:2".rpartition(':')
('1', ':', '2')
>>> "2".rpartition(':')
('', '', '2')
```
In the case wehere sep doesn't exist, the data ends up in port / the last chunk. From the docs:
"If the separator is not found, return a 3-tuple containing two empty strings, followed by the string itself." -- the first two end up empty.